Core/Interfaces/IRepository.cs

```diff
@@ -0,0 +1,15 @@
+﻿using LearnNet_CatalogService.Data.Common;
+using System.Linq.Expressions;
+
+namespace LearnNet_CatalogService.Core.Interfaces
+{
+    public interface IRepository<T> where T : BaseEntity
+    {
+        public Task<IQueryable<T>> GetAll();
+        public Task<T?> GetById(int id, params Expression<Func<T, object>>[] includes);
+        public Task Add(T entity);
+        public Task Update(T entity);
+        public Task Delete(int id);
+        public Task<int> Commit();
```

Comment:
I don't really see the benefit of this method in this case. IMO the code would be cleaner if repository just called SaveChanges() after each operation

Reply:
it depends, but in our simple crud - agree